pkg/api/v20240812preview/openshiftcluster.go

```diff
@@ -55,6 +58,9 @@ type OpenShiftClusterProperties struct {
 	// The cluster service principal profile.
 	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 
+	// The workload identity profile.
+	PlatformWorkloadIdentityProfile PlatformWorkloadIdentityProfile `json:"platformWorkloadIdentityProfile,omitempty"`
```

Comment:
This field is a pointer in the design doc to allow for it to be `nil` for CSP clusters.